bim2sim/tasks/bps/resolve_type_mismatch.py

```diff
@@ -0,0 +1,68 @@
+from bim2sim.tasks.base import ITask
+from bim2sim.utilities.common_functions import filter_elements
+from bim2sim.elements.bps_elements import (InnerWall, Floor, OuterWall,
+                                           GroundFloor, Roof)
+
+
+class ResolveTypeMismatch(ITask):
+    """Resolves issues with types, run() method holds detailed information."""
+    reads = ('elements', 'space_boundaries', 'disaggregations', 'tz_elements')
```

Comment:
You are reading all these elements and subsets of the elements, but you are only modifying tz_elements and their objects. 
Since you're not updating the others, your type-mismatch-resolving does not change anything on the EnergyPlus side. Could you please update the other lists also? Thanks!

Reply:
I think you got that wrong. The function changes the `elements` itself, it only gets the `elements` by iterating over the `thermalzone` instances ...
Maybe I misunderstood you, but please check your point again.

Reply:
We discussed this and this is not an issue with the implementation here but with the general concepts how we currently handle disaggregations, elements, tz_elements etc. We need a single source of truth here but this is part of another issue that will be written and tackled soon.